### PR TITLE
feat: force delete of appointments in delete doctor command

### LINKED ISSUE #75: delete/clear doctor bug
to fix after doctors are implemented

### PR DESCRIPTION
**Pull request by spzj**
Resolves issue #75 

When `delete-doctor --force INDEX` command is called on a doctor with existing appointments in the Appointment Schedule, the appointments will be mass deleted, along with the doctor.

### CODE REVIEW

File: src/main/java/seedu/address/commons/core/Messages.java [hunk 1]
```diff
@@ -13,19 +13,22 @@
...
+                    + " Force delete command format required! \n%1$s";
+    public static final String MESSAGE_CLEAR_APPOINTMENTS_BEFORE_PATIENTS_REQUIRED =
             "Patient records cannot be cleared until appointment schedule is cleared.\n"
-            + "Use the command \"clear-appt\" to clear the appointment schedule.";
-    public static final String MESSAGE_DELETE_PATIENT_SUCCESS = "Deleted Person: %1$s";
+                    + "Use the command \"clear-appt\" to clear the appointment schedule.";
+    public static final String MESSAGE_DELETE_PATIENT_SUCCESS = "Deleted Patient: %1$s";
 
-    //=================================Doctor related messages===============================================
+    //==================================Doctor related messages===============================================
```
Comment (onnwards) [inline src/main/java/seedu/address/commons/core/Messages.java]: this extra `=` misaligns it 😢

File: src/main/java/seedu/address/logic/parser/doctor/DeleteDoctorCommandParser.java [hunk 2]
```diff
@@ -14,17 +17,40 @@
 public class DeleteDoctorCommandParser implements Parser<DeleteDoctorCommand> {
 
     /**
-     * Parses the given {@code String} of arguments in the context of the DeleteDoctorCommand
+     * Used for the separation of force delete format and patient index from the format '--force INDEX'.
```
Comment (onnwards) [inline src/main/java/seedu/address/logic/parser/doctor/DeleteDoctorCommandParser.java]: javadocs has mention of patient

File: src/main/java/seedu/address/logic/parser/patient/DeletePatientCommandParser.java [hunk 3]
```diff
@@ -31,17 +31,17 @@ public DeletePatientCommand parse(String args) throws ParseException {
...
-                }
+        if (forceDeleteMatcher.matches()) {
+            final String forceDelete = forceDeleteMatcher.group("forceDelete").trim();
+            final String patientIndex = forceDeleteMatcher.group("patientIndex");
+
+            if (forceDelete.equals("--force")) {
+                isForceDelete = true;
+                indexToParse = patientIndex;
             }
+        }
```
Comment (onnwards) [inline src/main/java/seedu/address/logic/parser/patient/DeletePatientCommandParser.java:42]: actually, i feel like we should throw an exception if forceDelete not equals `"--force"` as well? feels like we shouldn't just let a user pass with an input such as `delete-patient aisjdhabakahbakshdbankdjashb 1`

if we do this we can then make the inner `if` conditional not nested, and more flat

same for `DeleteDoctorCommandParser`

**Review by onnwards — APPROVED** [review]
everything looks good, LGTM

File: src/main/java/seedu/address/logic/parser/doctor/DeleteDoctorCommandParser.java (hunk 2 above)
Comment (spzj) [reply src/main/java/seedu/address/logic/parser/doctor/DeleteDoctorCommandParser.java]: My bad, missed out on this one

File: src/main/java/seedu/address/commons/core/Messages.java (hunk 1 above)
Comment (spzj) [reply src/main/java/seedu/address/commons/core/Messages.java]: Meant to align the tail of the comment, will commit the changes.

File: src/main/java/seedu/address/logic/parser/patient/DeletePatientCommandParser.java (hunk 3 above)
Comment (spzj) [reply src/main/java/seedu/address/logic/parser/patient/DeletePatientCommandParser.java:42]: I don't think an exception is needed as `indexToParse = args` by default, so `aisjdhabakahbakshdbankdjashb 1` will be captured as a ParseException when `ParserUtil` calls `parseIndex` on it.

Comment (onnwards) [reply src/main/java/seedu/address/logic/parser/patient/DeletePatientCommandParser.java:42]: ah i missed that out, yeah then it works perfectly

**Pull request merged**